core/image.cpp

```diff
@@ -2521,6 +2521,10 @@ void Image::set_pixel(int p_x, int p_y, const Color &p_color) {
 	ERR_FAIL_INDEX(p_x, width);
 	ERR_FAIL_INDEX(p_y, height);
 
+	ERR_FAIL_COND(Math::is_nan(p_color.r * 255.0));
+	ERR_FAIL_COND(Math::is_nan(p_color.g * 255.0));
+	ERR_FAIL_COND(Math::is_nan(p_color.b * 255.0));
+	ERR_FAIL_COND(Math::is_nan(p_color.a * 255.0));
```

Comment:
Is there any situation where `x * 255.0` will be NaN if `x` is not NaN itself? If not, we could spare ourselves the multiplication.

I'm also not sure why we need to do this here. There are tons of methods taking floats as input and making operations with them, we can't possibly add NaN checks everywhere, we'd get 5 FPS in the editor...

Reply:
There is none. The only values for `x` in the expression `x * 255.0` that produce `NaN` is `NaN` itself. https://en.wikipedia.org/wiki/NaN#Operations_generating_NaN

Reply:
I just added this, because I got this error:
```
core/image.cpp:2552:31: runtime error: -nan is outside the range of representable values of type 'unsigned char'
```
I will try to find cause of existing of Nan in this variables to prevent this checks.